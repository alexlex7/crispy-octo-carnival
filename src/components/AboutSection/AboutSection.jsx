import React from 'react';
import PropTypes from 'prop-types';
import Container from '../Container/Container';
import FeatureList from '../FeatureList/FeatureList';
import styles from './AboutSection.module.css';

export default function AboutSection({ data }) {
  const { title, description, imageURL, mobileImageURL, features } = data;
  return (
    <Container>
      <div className={styles.wrapper}>
        <div className={styles.thumb}>
          <div className={styles.imageWrapper}>
            <img
              srcSet={`${mobileImageURL} 332w, ${imageURL} 596w`}
              sizes="(min-width: 1205px) 596px, (max-width: 1204px) 332px, 332px)"
              src={mobileImageURL}
              alt="furniture"
              title="furniture"
            />
          </div>
        </div>
        <div>
          <h2 className={styles.title}>{title}</h2>
          <p className={styles.text}>{description}</p>
          <FeatureList items={features} />
        </div>
      </div>
    </Container>
  );
}

AboutSection.propTypes = {
  data: PropTypes.shape({
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    imageURL: PropTypes.string.isRequired,
    mobileImageURL: PropTypes.string.isRequired,
    features: PropTypes.arrayOf(
      PropTypes.shape({
        title: PropTypes.string.isRequired,
        text: PropTypes.string.isRequired,
        imageURL: PropTypes.string.isRequired,
        mobileImageURL: PropTypes.string.isRequired,
      }).isRequired
    ),
  }).isRequired,
};
