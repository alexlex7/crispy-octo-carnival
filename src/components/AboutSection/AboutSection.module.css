.title {
  margin-top: 0;
  margin-bottom: 12px;

  font-family: 'Montserrat';
  font-style: normal;
  font-weight: 600;
  font-size: 24px;
  line-height: 40px;
  color: #244d4d;
}

.text {
  margin: 0px 0px 30px 0px;
  width: 322px;

  font-family: 'DM Sans';
  font-style: normal;
  font-weight: 400;
  font-size: 12px;
  line-height: 19px;
  color: #444444;
}

@media screen and (max-width: 1204px) {
  .wrapper {
    padding-left: 24px;
    display: flex;
    flex-direction: column-reverse;
    row-gap: 40px;
  }

  .thumb {
    position: relative;
    padding-top: 15px;
    padding-left: 14px;
  }

  .imageWrapper {
    width: 352px;
    height: 326px;
    background-color: #dee4e2;
  }

  .imageWrapper > img {
    display: block;
    position: relative;
    bottom: 15px;
  }
}

@media screen and (min-width: 1205px) {
  .wrapper {
    display: flex;
    column-gap: 100px;
    justify-content: center;
    align-items: center;
  }

  .thumb {
    padding-top: 26px;
  }

  .imageWrapper {
    width: 631px;
    height: 586px;
    background-color: #dee4e2;
  }

  .imageWrapper > img {
    display: block;
    position: relative;
    bottom: 26px;
  }

  .title {
    margin-bottom: 20px;
    font-weight: 700;
    font-size: 32px;
    line-height: 40px;
  }

  .text {
    margin-bottom: 40px;
    width: 400px;
    font-size: 20px;
    line-height: 25px;
  }
}
