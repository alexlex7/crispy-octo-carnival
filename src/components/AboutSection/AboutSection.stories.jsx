import React from 'react';
import AboutSection from './AboutSection';
import data from '../../mock/features.json';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'About/AboutSection',
  component: AboutSection,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
};

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template = (args) => <AboutSection {...args} />;

export const Default = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Default.args = {
  data,
};
