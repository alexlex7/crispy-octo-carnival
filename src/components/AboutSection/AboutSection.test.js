import { render, screen } from '@testing-library/react';
import AboutSection from './AboutSection';
import data from '../../mock/features.json';

test('render about section', () => {
  const { asFragment } = render(<AboutSection data={data} />);
  const title = screen.getByText(data.title);
  expect(title).toBeInTheDocument();

  const list = screen.getByRole('list');
  expect(list).toBeInTheDocument();

  expect(asFragment()).toMatchSnapshot();
});
