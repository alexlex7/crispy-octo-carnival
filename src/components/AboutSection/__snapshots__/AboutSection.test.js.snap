// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render about section 1`] = `
<DocumentFragment>
  <div
    class="container "
  >
    <div
      class="wrapper"
    >
      <div
        class="thumb"
      >
        <div
          class="imageWrapper"
        >
          <img
            alt="furniture"
            sizes="(min-width: 1205px) 596px, (max-width: 1204px) 332px, 332px)"
            src="http://localhost:8080/images/about/mobileAbout.jpg"
            srcset="http://localhost:8080/images/about/mobileAbout.jpg 332w, http://localhost:8080/images/about/about.jpg 596w"
            title="furniture"
          />
        </div>
      </div>
      <div>
        <h2
          class="title"
        >
          About Us
        </h2>
        <p
          class="text"
        >
          All of our furniture uses the best materials and choices for our customers.All of our furniture uses the best materials
        </p>
        <ul
          class="list"
        >
          <li
            class="listItem"
          >
            <div
              class="thumb"
            >
              <img
                alt="Best Quality"
                sizes="(min-width: 1205px) 50px, (max-width: 1204px) 28px, 28px)"
                src="http://localhost:8080/images/features/mobileQuality.png"
                srcset="http://localhost:8080/images/features/mobileQuality.png 28w, http://localhost:8080/images/features/quality.png 50w"
                title="Best Quality"
              />
            </div>
            <div>
              <h6
                class="title"
              >
                Best Quality
              </h6>
              <p
                class="text"
              >
                All of our furniture uses the best materials and choices
              </p>
            </div>
          </li>
          <li
            class="listItem"
          >
            <div
              class="thumb"
            >
              <img
                alt="Free Shipping"
                sizes="(min-width: 1205px) 50px, (max-width: 1204px) 28px, 28px)"
                src="http://localhost:8080/images/features/mobileBest.png"
                srcset="http://localhost:8080/images/features/mobileBest.png 28w, http://localhost:8080/images/features/best.png 50w"
                title="Free Shipping"
              />
            </div>
            <div>
              <h6
                class="title"
              >
                Free Shipping
              </h6>
              <p
                class="text"
              >
                All of our furniture uses the best materials and choices
              </p>
            </div>
          </li>
          <li
            class="listItem"
          >
            <div
              class="thumb"
            >
              <img
                alt="Free Shipping"
                sizes="(min-width: 1205px) 50px, (max-width: 1204px) 28px, 28px)"
                src="http://localhost:8080/images/features/mobileShipping.png"
                srcset="http://localhost:8080/images/features/mobileShipping.png 28w, http://localhost:8080/images/features/shipping.png 50w"
                title="Free Shipping"
              />
            </div>
            <div>
              <h6
                class="title"
              >
                Free Shipping
              </h6>
              <p
                class="text"
              >
                All of our furniture uses the best materials and choices
              </p>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
